staging/src/k8s.io/client-go/tools/events/event_broadcaster.go

```diff
@@ -203,9 +203,9 @@ func recordEvent(sink EventSink, event *v1beta1.Event) (*v1beta1.Event, bool) {
 	var err error
 	isEventSeries := event.Series != nil
 	if isEventSeries {
-		patch, err := createPatchBytesForSeries(event)
-		if err != nil {
-			klog.Errorf("Unable to calculate diff, no merge is possible: %v", err)
+		patch, patchBytesErr := createPatchBytesForSeries(event)
```

Comment:
I don't see why the existing code would prevent checking the patch/create err ? (it is checked [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/event_broadcaster.go#L219))

Reply:
@yastij
The bug was due to the assignment to err declared on line 206 which would go out of scope coming out of the if block.
The check you cited wouldn't be applied on the desired err variable.

Reply:
sgtm